iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -549,16 +550,27 @@ private void commitOverwrite(Table table, String branchName, long startTime, Fil
     if (!results.dataFiles().isEmpty()) {
       Transaction transaction = table.newTransaction();
       if (rewritePolicy == RewritePolicy.ALL_PARTITIONS) {
-        DeleteFiles delete = transaction.newDelete();
-        delete.deleteFromRowFilter(Expressions.alwaysTrue());
-        delete.commit();
-      }
-      ReplacePartitions overwrite = transaction.newReplacePartitions();
-      results.dataFiles().forEach(overwrite::addFile);
-      if (StringUtils.isNotEmpty(branchName)) {
-        overwrite.toBranch(HiveUtils.getTableSnapshotRef(branchName));
+
+        List<DataFile> existingDataFiles = Lists.newArrayList();
+        List<DeleteFile> existingDeleteFiles = Lists.newArrayList();
+        IcebergTableUtil.getFiles(table, existingDataFiles, existingDeleteFiles);
```

Comment:
Can we reuse `FilesForCommit results` to get the dataFiles and deleteFiles?

Reply:
I don't think so, because FilesForCommit contains only the compacted, new files. They don't contain existing data and delete files.

Reply:
Since the Major QB Compaction is essentially a `IOW `operation, why don't we follow the regular `IOW `flow?  Why do we need the extra operation to find the existing data and delete files? 
Just reuse the  `ReplacePartitions  `api:
https://github.com/apache/hive/blob/e08a60029f93e52182908b61fff32e8659e78daa/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java#L556-L561

Reply:
Because IOW isn't supported on tables that had schema evolution as per https://issues.apache.org/jira/browse/HIVE-26133, but we want to support compaction even for tables that have had schema evolution.

Reply:
Got it, thx.

It seems that we have tradeoff about the change.
(twice commit) vs (single commit: loop to get and store all existing data and delte files)

I know twice commit will generate two snapshot and maybe get inconsistency of data when users reading table.
But for this change, if the original iceberg table has many many small data and delete files, and we need to loop and store these files in the `List`,  can this behavior would give memory pressure on HS?
I am not sure which way is the best one. ;(

Can other folks give some other thought? @deniskuzZ @ayushtkn @SourabhBadhya